src/utils/array-utils.js

```diff
@@ -22,6 +22,17 @@ export function removeItemAt(items, index) {
   return items.filter((t, i) => i !== index);
 }
 
+export function replaceItemAt(items, index, newItem) {
+  if (!items || index < 0 || index > (items.length - 1)) {
```

Comment:
I'd prefer the index >= items.length since you don't need the 
parenthesis and the extra math

Reply:
True. From meaning perspective though it makes more sense to check for 'before the first' or 'after the last' item in the string. Would leave it and boil it down to taste 😄

Reply:
Ok, I have a mathy tastet :)